app/src/main/java/com/example/drawingactivity/HueView.kt

```diff
@@ -0,0 +1,98 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+
+@SuppressLint("ClickableViewAccessibility")
+class HueView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    var onHueChanged: ((Int) -> Unit)? = null
+    var onSliderClicked: ((Boolean) -> Unit)? = null
+    private var hueSlider = 360f
+    private var currentHue: Int = 0
+    private var sliderBitmap: Bitmap? = null
+    private var dy = hueSlider
+    private val selectPaint = Paint().apply {
+        strokeWidth = 12f
+        color = Color.BLACK
+        style = Paint.Style.STROKE
+    }
+
+    init {
+        setOnTouchListener { _, mEvent ->
+            dy = ((mEvent.y / height) * 360).coerceIn(0f, 360f)
+            when (mEvent.action) {
+                MotionEvent.ACTION_DOWN -> {
+                    onSliderClicked?.invoke(true)
+                    setHue(dy.toInt())
+                    pickHue()
+                    invalidate()
+                }
+                MotionEvent.ACTION_MOVE -> {
+                    setHue(dy.toInt())
+                    pickHue()
+                    invalidate()
+                }
+                MotionEvent.ACTION_UP -> {
+                    onSliderClicked?.invoke(false)
+                }
+            }
+            true
+        }
+    }
+
+    override fun onDraw(canvas: Canvas?) {
+        super.onDraw(canvas)
+        canvas?.apply {
+            if (sliderBitmap == null) {
+                sliderBitmap = makeBitmap()
+            }
+            sliderBitmap?.let { drawBitmap(it, 0f, 0f, null) }
+            drawLine(0f, (dy / 360) * height, width * 1f, (dy / 360) * height, selectPaint)
+        }
+    }
+
+    private fun pickHue() {
+        onHueChanged?.invoke(currentHue)
+        invalidate()
+    }
+
+    fun setHue(newHue: Int) {
+        currentHue = newHue
+        dy = newHue.toFloat().coerceIn(0f, 360f)
```

Comment:
Ctrl+Shift+F for 360 and constant them lol

Reply:
Made constants